### PR TITLE
Refactoring of #8369

### LINKED ISSUE #8369: Column: Add converter attribute for filter and short notation
So you can do
```xhtml
<p:column headerText="ID lt" field="id" filterMatchMode="lt" converter="javax.faces.Integer"/>
```
instead of
```xhtml
<p:column headerText="ID lt" field="id" filterMatchMode="lt">
    <f:facet name="filter">
         <p:inputText onkeyup="PF('wgtTable').filter()">
              <f:converter converterId="javax.faces.Integer" />
          </p:inputText>
     </f:facet>
</p:column>
```

### PR DESCRIPTION
**Pull request by Rapster**
Fix #8369

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/columns/ColumnsBase.java [hunk 1]
```diff
@@ -475,4 +476,12 @@ public void setTitle(String title) {
         getStateHelper().put(PropertyKeys.title, title);
     }
 
+    @Override
+    public String getConverter() {
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/columns/ColumnsBase.java]: Why string here and not object?

Comment (Rapster) [reply primefaces/src/main/java/org/primefaces/component/columns/ColumnsBase.java]: bad copy/paste :D

File: primefaces/src/main/java/org/primefaces/component/api/UITable.java [hunk 2]
```diff
@@ -203,6 +204,18 @@ default void updateFilterByValuesWithFilterRequest(FacesContext context, Map<Str
                 filterValue = params.get(valueHolderClientId);
             }
 
+            if (filterMeta.isShortFieldNotation()) {
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/api/UITable.java]: can you add a comment here, why its only done when #isShortFieldNotiation?

everything else looks good -> +1

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
do we need a taglib change?

**Pull request merged**